src/Uno.UI.Runtime.Skia.Gtk/Input/GtkKeyboardInputSource.cs

```diff
@@ -80,15 +83,60 @@ private void OnKeyReleaseEvent(EventKey evt)
 					{
 						ScanCode = evt.HardwareKeycode,
 						RepeatCount = 1,
-					}));
+					},
+					KeyCodeToUnicode(evt.HardwareKeycode, evt.KeyValue)));
 		}
 		catch (Exception e)
 		{
 			Windows.UI.Xaml.Application.Current.RaiseRecoverableUnhandledException(e);
 		}
 	}
 
-	private VirtualKey ConvertKey(Gdk.Key key)
+	private static char? KeyCodeToUnicode(uint keyCode, uint keyVal)
+	{
+		if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+		{
+			var result = WindowsKeyCodeToUnicode(keyCode);
+			return result.Length > 0 ? result[0] : null; // TODO: supplementary code points
+		}
+
+		var gdkChar = (char)Keyval.ToUnicode(keyVal);
+
+		return gdkChar == 0 ? null : gdkChar;
+	}
+
+	[DllImport("user32.dll")]
```

Comment:
Could the Windows-specific logic be abstracted away so that the implementation can be shared between GTK and WPF? There are other pieces that would benefit from it, e.g. `DpiHelper` I think.

Reply:
Yes, I just didn't know where to put that shared logic.

Reply:
@jeromelaban What do you think? Would it be better to have a separate library or just link the file from the other project?

Reply:
Let's link the file for now. Having a separate library is not worth it for now.